### PR TITLE
STORY#551 Completed

### PR DESCRIPTION
**Pull request by rejneesh1**

### CODE REVIEW

File: src/pages/tutorial-video-listing/tutorial-video-listing.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
+
+<ion-header>
+  <header [title]="'headings.tutorialvideo'|translate" [hideBack]="false">
+  </header>
+</ion-header>
+
+
+<ion-content>
+
+  <ion-refresher slot="fixed" (ionRefresh)="refresh($event)" *ngIf="enableRefresh">
```
Comment (kiranharidas187) [inline src/pages/tutorial-video-listing/tutorial-video-listing.html]: Why do we need a refresher here?

File: src/pages/tutorial-video-listing/tutorial-video-listing.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  ionViewDidLoad(){
+    this.localStorage.getLocalStorage(storageKeys.staticLinks).then(success => {
+      this.videoList=success['tutorial-video'].metaInformation.videos
+    }).catch(error => {
+    })
+  }
+ 
+
+  playVideo(){
+    // this.youtube.openVideo('ovqDe_G7ct8');
```
Comment (kiranharidas187) [inline src/pages/tutorial-video-listing/tutorial-video-listing.ts:32]: remove this id

File: src/app/app.component.ts [hunk 3]
```diff
@@ -181,6 +190,12 @@ export class MyApp {
 
     });
 
+    this.localStorage.getLocalStorage(storageKeys.staticLinks).then(success => {
+     console.log(success)
+     if(success['tutorial-video'].metaInformation.videos.length==0)this.allPages[this.findIndex("tutorialvideo")]['show']=false
```
Comment (kiranharidas187) [inline src/app/app.component.ts]: will this work??

**Review by kiranharidas187 — COMMENTED** [review]
Please resolve this

File: src/pages/tutorial-video-listing/tutorial-video-listing.ts (hunk 2 above)
Comment (rejneesh1) [reply src/pages/tutorial-video-listing/tutorial-video-listing.ts:32]: > remove this id

removed and passing the link in function

File: src/pages/tutorial-video-listing/tutorial-video-listing.html (hunk 1 above)
Comment (rejneesh1) [reply src/pages/tutorial-video-listing/tutorial-video-listing.html]: > Why do we need a refresher here?

removed

File: src/pages/tutorial-video-listing/tutorial-video-listing.ts (hunk 2 above)
Comment (rejneesh1) [reply src/pages/tutorial-video-listing/tutorial-video-listing.ts:32]: Above id is video id we are not using this so commented

File: src/app/app.component.ts (hunk 3 above)
Comment (rejneesh1) [reply src/app/app.component.ts]: yes i tested it with length ==1 as we already have one video so it was hiding in side menu ..it should work with length ==0  also and should hide  from side menu . tried after logout also. it was working

File: src/pages/tutorial-video-listing/tutorial-video-listing.html [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            <ion-icon name="play" class="menuIcon" ></ion-icon>   
+          </div>
+        </ion-item>
+      </ion-card>
+    </div>
+  </div>
+
+
+  <!-- if video not present -->
+  <div *ngIf="videoList?.length==0">
```
Comment (kiranharidas187) [inline src/pages/tutorial-video-listing/tutorial-video-listing.html]: use !videoList?.length, its the same as videoList?.length==0

File: src/pages/tutorial-video-listing/tutorial-video-listing.html [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    <div class="ion-diagnostics-sheet-title">
+      <h3>No videos available</h3>
```
Comment (kiranharidas187) [inline src/pages/tutorial-video-listing/tutorial-video-listing.html]: should use translate service

File: src/pages/tutorial-video-listing/tutorial-video-listing.html [hunk 6]
```diff
@@ -0,0 +1,34 @@
+
+<ion-header>
+  <header [title]="'headings.tutorialvideo'|translate" [hideBack]="false">
+  </header>
+</ion-header>
+
+
+<ion-content >
+  <!-- if video presenet -->
+  <div *ngIf="videoList?.length>0">
```
Comment (kiranharidas187) [inline src/pages/tutorial-video-listing/tutorial-video-listing.html]: videoList?.length will also return true

File: src/pages/tutorial-video-listing/tutorial-video-listing.html (hunk 4 above)
Comment (rejneesh1) [reply src/pages/tutorial-video-listing/tutorial-video-listing.html]: changed

File: src/pages/tutorial-video-listing/tutorial-video-listing.html (hunk 5 above)
Comment (rejneesh1) [reply src/pages/tutorial-video-listing/tutorial-video-listing.html]: using translate service now to show "No videos available"

File: src/pages/tutorial-video-listing/tutorial-video-listing.html (hunk 6 above)
Comment (rejneesh1) [reply src/pages/tutorial-video-listing/tutorial-video-listing.html]: ok changed

**Pull request merged**